### DISCUSSION TITLE
Instructions for multi platform development

### DISCUSSION THREAD

**Discussion by dapalmi** [dotnet/dotnet-docker#4430] [Q&A]
Thanks a lot for the documentation how to use the new and easy to use HTTPS by default for Asp.Net Core 2.1 with Docker containers (https://github.com/dotnet/dotnet-docker/blob/master/samples/aspnetapp/aspnetcore-docker-https.md).

I have two request to make development easier:

1. Because of %USERPROFILE% (Windows) and ${HOME} (Mac) in the docker-compose file it is not useful if this file is used both for Mac developers and Windows developers. Is there a way to make this multi platform compatible?

2. Is there a way to execute those commands when executing the docker-compose file to check if a certificate already exists and to create one if it does not yet exists:
dotnet dev-certs https -ep %USERPROFILE%\.aspnet\https\aspnetapp.pfx -p crypticpassword
dotnet dev-certs https --trust

**Comment by MichaelSimons** [comment]
@dapalmi - Thanks for the feedback on the samples!

**Comment by mthalman** [comment]
> Because of %USERPROFILE% (Windows) and ${HOME} (Mac) in the docker-compose file it is not useful if this file is used both for Mac developers and Windows developers. Is there a way to make this multi platform compatible?

@dapalmi - As far as environment variables go, you should be able to define your docker-compose file in a multi-platform manner by using the `${VARIABLE}` syntax.  See the [Docker Compose documentation](https://docs.docker.com/compose/compose-file/#service-configuration-reference): _You can use environment variables in configuration values with a Bash-like `${VARIABLE}` syntax - see variable substitution for full details._

> Is there a way to execute those commands when executing the docker-compose file to check if a certificate already exists and to create one if it does not yet exists:
dotnet dev-certs https -ep %USERPROFILE%.aspnet\https\aspnetapp.pfx -p crypticpassword
dotnet dev-certs https --trust

That's essentially what these commands do already.  If the cert already exists, it doesn't create it.  Is that sufficient for you?  How is this relevant to the docker-compose file?

**Comment by gmartinezsan** [comment]
hi all, 

Adding to the suggestion of @dapalmi, I have also found myself looking to have a way to run these commands in the container. Currently, the process requires generating the certificate in the host, or at least that is what the docs say., which makes it a manual process. But if we can run them in the docker-compose, it can be automated. In order to execute the commands, there is a dependency on the SDK. Is there a way to run these commands either in the docker-compose or docker file, but perhaps is not documented?

Thanks in advance,

**Comment by mthalman** [comment]
[Triage]
Based on the ask here, it sounds like you're looking for guidance on how to make use of .NET certificates in the context of docker-compose and to do so in a way that is compatible with multiple platforms. Since this isn't strictly an issue with the .NET Docker images, this issue will be converted to a discussion so that others in the community can help out.
